cluster-state-service/handler/store/instancestore.go

```diff
@@ -130,19 +134,51 @@ func (instanceStore eventInstanceStore) ListContainerInstances() ([]types.Contai
 }
 
 // FilterContainerInstances returns all container instances from the datastore that match the provided filters
-func (instanceStore eventInstanceStore) FilterContainerInstances(filterKey string, filterValue string) ([]types.ContainerInstance, error) {
-	if len(filterKey) == 0 || len(filterValue) == 0 {
-		return nil, errors.New("Filter key and value cannot be empty")
+func (instanceStore eventInstanceStore) FilterContainerInstances(filterMap map[string]string) ([]types.ContainerInstance, error) {
+	if len(filterMap) == 0 {
+		return nil, errors.New("There has to be at least one filter")
+	}
+
+	filters := make([]string, 0, len(filterMap))
+	for k := range filterMap {
+		filters = append(filters, k)
+	}
+
+	if !instanceStore.areFiltersValid(filters) {
+		return nil, errors.Errorf("At least one of the provided filters '%v' is not supported.", filters)
+	}
+
+	for _, f := range filters {
+		if filterMap[f] == "" {
+			return nil, errors.Errorf("Filter value for filter '%s' is empty", f)
+		}
 	}
 
-	switch {
-	case filterKey == instanceStatusFilter:
-		return instanceStore.filterContainerInstancesByStatus(filterValue)
-	case filterKey == instanceClusterFilter:
-		return instanceStore.filterContainerInstancesByCluster(filterValue)
-	default:
-		return nil, errors.Errorf("Unsupported filter key: %s", filterKey)
+	if len(filters) == 1 {
+		filterKey := filters[0]
+		filterValue := filterMap[filterKey]
+		switch {
+		case filterKey == instanceStatusFilter:
+			return instanceStore.filterContainerInstancesByStatus(filterValue)
+		case filterKey == instanceClusterFilter:
+			return instanceStore.filterContainerInstancesByCluster(filterValue)
+		}
+	}
+
+	// Currently the only combination of filters that is supported are status & cluster
+	if len(filters) == 2 {
+		status, ok := filterMap[instanceStatusFilter]
+		if !ok {
```

Comment:
Could this be simplified as `cluster, ok2 := filterMap[instanceClusterFilter]` and combining the two checks into one `if !ok || !ok2 {`?